src/MizarData/FrameTrackManagerTest.cpp

```diff
@@ -0,0 +1,233 @@
+    EXPECT_CALL(capture_data_, GetScopeInfo).WillRepeatedly(Invoke([](const ScopeId id) {
+      return kScopeIdToInfo.at(id);
+    }));
+    EXPECT_CALL(capture_data_, GetTimersForScope)
+        .WillRepeatedly(Invoke([](const ScopeId id, uint64_t /*min*/, uint64_t /*max*/) {
+          return kScopeIdToSTimerInfoPtrs.at(id);
+        }));
+  }
+
+  void ExpectGetFrameStartsIsCorrect(FrameStartNs min_start, FrameStartNs max_start) {
+    for (const auto& [id, info] : frame_track_manager_.GetFrameTracks()) {
+      const std::vector<FrameStartNs> expected = Visit(
+          [](const ScopeInfo& info) { return kScopeInfoToFrameStarts.at(info); },
+          [min_start, max_start](PresentEvent::Source source) {
+            std::vector<FrameStartNs> expected;
+            const std::vector<FrameStartNs> all_frame_starts = kEtwSourceToFrameStart.at(source);
+            std::copy_if(std::begin(all_frame_starts), std::end(all_frame_starts),
+                         std::back_inserter(expected), [min_start, max_start](FrameStartNs start) {
+                           return min_start <= start && start <= max_start;
+                         });
+            return expected;
+          },
+          info);
+
+      const std::vector<FrameStartNs> actual =
+          frame_track_manager_.GetFrameStarts(id, min_start, max_start);
+      EXPECT_THAT(actual, UnorderedElementsAreArray(expected));
+    }
+  }
+
+  void ExpectGetFrameTracksIsCorrect(
+      const std::vector<orbit_client_data::ScopeInfo>& expected_scope_info,
+      const std::vector<PresentEvent::Source>& expected_etw_sources) {
+    const auto [scope_infos, etw_sources] = DecomposeSources(frame_track_manager_.GetFrameTracks());
+
+    EXPECT_THAT(scope_infos, UnorderedElementsAreArray(expected_scope_info));
+    EXPECT_THAT(etw_sources, UnorderedElementsAreArray(expected_etw_sources));
+  }
+
+ protected:
+  MockCaptureData capture_data_;
+  MockMizarData data_;
+  FrameTrackManagerTmpl<MockMizarData> frame_track_manager_{&data_};
+};
+
+TEST_F(ExpectFrameTracksHasFrameStartsForScopes, FrameTracksAreCorrectForNoScopesAndNoEtws) {
+  const absl::flat_hash_map<FrameTrackId, FrameTrackInfo> id_to_infos =
+      frame_track_manager_.GetFrameTracks();
+
+  EXPECT_THAT(id_to_infos, IsEmpty());
+}
+
+TEST_F(ExpectFrameTracksHasFrameStartsForScopes, FrameTracksAreCorrectForScopesAndNoEtws) {
+  CaptureDataExpectCalls();
+
+  ExpectGetFrameTracksIsCorrect(kScopeInfos, {});
+
+  // Filtering of scopes w.r.t min/max timestamp is handles by Capture data, it is not covered
+  // in the test, hence we just pass zeroes. The MockCaptureData ignores these anyway.
+  ExpectGetFrameStartsIsCorrect(FrameStartNs(0), FrameStartNs(0));
+}
+
+TEST_F(ExpectFrameTracksHasFrameStartsForScopes, FrameTracksAreCorrectForEtwsAndNoScopes) {
+  SourceToPresentEventExpectCall();
+
+  ExpectGetFrameTracksIsCorrect({}, kEtwSources);
+
+  ExpectGetFrameStartsIsCorrect(FrameStartNs(0), FrameStartNs(15));
```

Comment:
These constants doesn't have sense without the initial context, please add a comment explaining why you are choosing 0,15,50 and 300